src/Simplex/Chat.hs

```diff
@@ -5145,10 +5154,13 @@ chatCommandP =
       (">#" <|> "> #") *> (SendGroupMessageQuote <$> displayName <* A.space <* char_ '@' <*> (Just <$> displayName) <* A.space <*> quotedMsg <*> msgTextP),
       "/_contacts " *> (APIListContacts <$> A.decimal),
       "/contacts" $> ListContacts,
-      "/_connect " *> (APIConnect <$> A.decimal <* A.space <*> ((Just <$> strP) <|> A.takeByteString $> Nothing)),
-      "/_connect " *> (APIAddContact <$> A.decimal),
-      ("/connect " <|> "/c ") *> (Connect <$> ((Just <$> strP) <|> A.takeByteString $> Nothing)),
-      ("/connect" <|> "/c") $> AddContact,
+      "/_connect " *> (APIConnect <$> A.decimal <* " incognito=" <*> onOffP <* A.space <*> ((Just <$> strP) <|> A.takeByteString $> Nothing)),
+      "/_connect " *> (APIAddContact <$> A.decimal <* " incognito=" <*> onOffP),
+      "/_set incognito :" *> (APISetConnectionIncognito <$> A.decimal <* A.space <*> onOffP),
+      ("/connect incognito " <|> "/c i ") *> (Connect True <$> ((Just <$> strP) <|> A.takeByteString $> Nothing)),
+      ("/connect incognito" <|> "/c i") $> AddContact True,
+      ("/connect " <|> "/c ") *> (Connect False <$> ((Just <$> strP) <|> A.takeByteString $> Nothing)),
+      ("/connect" <|> "/c") $> AddContact False,
```

Comment:
it would be simpler to parse incognito separately
```suggestion
      ("/connect " <|> "/c ") *> (Connect <$> incognitoP <*> ((Just <$> strP) <|> A.takeByteString $> Nothing)),
      ("/connect " <|> "/c ") *> AddContact <$> incognitoP,
```

where: `incognitoP = ((" incognito" <|> " I") $> True <|> pure False)`